From 7ba515e4dd7519ebfcec127d785ffc87cf013136 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 11 Oct 2023 16:51:15 +0200
Subject: [PATCH 0188/1695] clk: stm32mp25: add CLK_SET_RATE_PARENT flag for
 LTDC and STGEN clocks

Transmit change rate on Flexgen clocks for these clocks.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2ef7994c074406206e8ff98767a761dcb427a0ce clk: stm32mp25: add CLK_SET_RATE_PARENT flag for LTDC and STGEN clocks]
Change-Id: Ia7a9a3e7cc129c4a3002d12be4be906c9b7e5e9c
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/drivers/clk/clk-stm32mp25.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index 6d42b7ae2..35bf2c54b 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -2663,7 +2663,8 @@ static STM32_GATE(ck_ker_fdcan, &ck_flexgen_26, 0, GATE_FDCAN);
 static STM32_GATE(ck_ker_csi2, &ck_flexgen_29, 0, GATE_CSI);
 static STM32_GATE(ck_ker_csi2txesc, &ck_flexgen_30, 0, GATE_CSI);
 static STM32_GATE(ck_ker_csi2phy, &ck_flexgen_31, 0, GATE_CSI);
-static STM32_GATE(ck_ker_stgen, &ck_flexgen_33, 0, GATE_STGEN);
+static STM32_GATE(ck_ker_stgen, &ck_flexgen_33, CLK_SET_RATE_PARENT,
+		  GATE_STGEN);
 static STM32_GATE(ck_ker_usbtc, &ck_flexgen_35, 0, GATE_USBTC);
 static STM32_GATE(ck_ker_i3c4, &ck_flexgen_36, 0, GATE_I3C4);
 static STM32_GATE(ck_ker_spi8, &ck_flexgen_37, 0, GATE_SPI8);
@@ -2696,7 +2697,8 @@ static STM32_GATE(ck_ker_ethswref, &ck_flexgen_60, 0, GATE_ETHSWREF);
 static STM32_GATE(ck_ker_eth1stp, &ck_icn_ls_mcu, 0, GATE_ETH1STP);
 static STM32_GATE(ck_ker_eth2stp, &ck_icn_ls_mcu, 0, GATE_ETH2STP);
 
-static STM32_GATE(ck_ker_ltdc, &ck_flexgen_27, 0, GATE_LTDC);
+static STM32_GATE(ck_ker_ltdc, &ck_flexgen_27, CLK_SET_RATE_PARENT,
+		  GATE_LTDC);
 
 static STM32_COMPOSITE(ck_mco1, 2, PARENT(&ck_flexgen_61, &ck_obser0),
 		       0, GATE_MCO1, NO_DIV, MUX_MCO1);
-- 
2.47.2

